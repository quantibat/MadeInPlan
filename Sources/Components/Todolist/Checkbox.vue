<template id="Checkbox">
    <div id="_prevent" :class="'switch ' + (done ? 'true' : 'false' )" @click="handleToggle()">
        <span id="_prevent"></span>
    </div>
</template>

<script>
export default {
    name: 'Checkbox',
    props: {
        'id': {
            type: String,
            required: true
        },
        'done': {
            type: Boolean,
            required: true
        },
        'cb': {
            type: Function,
            required: false
        }
    },
    data() {
        return {
            messages: ["Bravo ðŸ˜€", "GÃ©nial !", "Top !", "Une bonne chose de faite !",
            "Quel talent !", "Ã‡a câ€™est super !", "Super !", "Excellent !", "Topissime !", "Sans pitiÃ© !",
            "Vamos !", "Cool !", "Vous avancez bien, bravo !", "Fantastique !", "Câ€™est bon Ã§a !",
            "VoilÃ  une bonne nouvelle !", "Yes ! Rien ne vous arrÃªte !", "Parfait ! Ne changez rien !",
            "En avant !", "En avant Guingamp !", "Vous assurez !"]
        }
    },
    computed: {
        tasklist() {
            return this.$store.state.tasklist;
        }
    },
    methods: {
        handleToggle() {
            if (this.tasklist.length > 0) this.$store.commit('toggleTask', this.id);
            
            if (!this.done) {
                this.$message({
                    message: this.messages[Math.floor(Math.random() * Math.floor(this.messages.length))],
                    type: 'success',
                    customClass: 'toogle-checkbox-message-class'
                });
            }
            this.$http.post(this.$root.URL + '/todolist/toggle-task', { id: this.id }).then(res => {
                if (this.cb) this.cb(this.id);
            }, res => {
                console.log('ERROR toggleTask Server');
            });
        }
    }
}
</script>

<style scoped>
.switch::before {
    content: '';
    position: absolute;
    left: -20px;
    top: -20px;
    width: 70px;
    height: 85px;
    background: transparent;
}

.switch {
    position: absolute;
    width: 42px;
    height: 42px;
    text-align: center;
    margin: 0;
    margin-top: 5px;
    background: #2ECC71;
    background: linear-gradient(5deg, rgba(46,204,113,1) 0%, rgba(145,212,101,1) 100%);
    -webkit-transition: all 0.2s ease;
    -moz-transition: all 0.2s ease;
    -o-transition: all 0.2s ease;
    -ms-transition: all 0.2s ease;
    transition: all 0.2s ease;
    border-radius: 25px;
}
.switch span {
    position: absolute;
    width: 19px;
    height: 3px;
    top: 50%;
    left: 50%;
    margin: -2px 0px 0px -7px;
    background: #FFFFFF;
    display: block;
    -webkit-transform: rotate(-45deg);
    -moz-transform: rotate(-45deg);
    -o-transform: rotate(-45deg);
    -ms-transform: rotate(-45deg);
    transform: rotate(-45deg);
    -webkit-transition: all 0.2s ease;
    -moz-transition: all 0.2s ease;
    -o-transition: all 0.2s ease;
    -ms-transition: all 0.2s ease;
    transition: all 0.2s ease;
    border-radius: 25px;
}
.switch span::after {
    content: "";
    display: block;
    position: absolute;
    width: 3px;
    height: 9px;
    margin: -6px 0px 0px 0px;
    background: #FFFFFF;
    -webkit-transition: all 0.2s ease;
    -moz-transition: all 0.2s ease;
    -o-transition: all 0.2s ease;
    -ms-transition: all 0.2s ease;
    transition: all 0.2s ease;
    border-radius: 25px;
}
.false {
    width: 38px;
    height: 38px;
    background: #FFFFFF;
    border: solid 2px #E0E0E0;
}
.false span {
    background: #E0E0E0;
}
.false span::after {
    background: #E0E0E0;
}

.false:hover { border-color: #2ECC71; }
.false:hover span { background: #2ECC71; background: linear-gradient(5deg, rgba(46,204,113,1) 0%, rgba(145,212,101,1) 100%); }
.false:hover span::after { background: #2ECC71; background: linear-gradient(5deg, rgba(46,204,113,1) 0%, rgba(145,212,101,1) 100%);}
</style>
