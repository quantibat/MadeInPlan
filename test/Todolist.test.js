import Vue from 'vue'
import Vuex from 'vuex';
import { mount, shallowMount, createLocalVue } from '@vue/test-utils'
import Checkbox from '../Sources/Components/Todolist/Checkbox.vue'
import { __createMocks as createStoreMocks, mockRequest } from './_mocks_/Todolist.mock.js';
import ElementUI from 'element-ui';
import locale from 'element-ui/lib/locale/lang/fr';
import axios from 'axios'
const localVue = createLocalVue();

Vue.use(ElementUI, { locale });
localVue.use(ElementUI, { locale });
Vue.use(Vuex);
localVue.use(Vuex);


describe('Todolist', () => {
    describe('Checkbox', () => {    
        beforeEach(() => {
            storeMocks = createStoreMocks();
            wrapper = shallowMount(Checkbox, {
                propsData: {
                    id: "ac7a758514e87136a17c",
                    done: false,
                    cb: id => id
                },
                store: storeMocks.store,
                mutations: storeMocks.mutations,
                localVue,
            });
            Vue.prototype.$http = mockRequest;
            localVue.prototype.$http = mockRequest;
            store = new Vuex.Store(storeMocks)
        });
    
        it('renders correctly', () => {
            expect(wrapper.html()).toMatchSnapshot("checkbox");
        });

        it('Calls ToggleTask when checked', () => {
            wrapper.find('.switch').trigger('click');
            expect(storeMocks.mutations.toggleTask).toBeCalled();
        });
    });
});
