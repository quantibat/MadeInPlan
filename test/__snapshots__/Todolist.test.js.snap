// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todolist Checkbox renders correctly: checkbox 1`] = `"<div id=\\"_prevent\\" class=\\"switch false\\"><span id=\\"_prevent\\"></span></div>"`;
